central/role/datastore/datastore_impl.go

```diff
@@ -126,14 +126,20 @@ func (ds *dataStoreImpl) RemoveRole(ctx context.Context, name string) error {
 	if err := sac.VerifyAuthzOK(roleSAC.WriteAllowed(ctx)); err != nil {
 		return err
 	}
-	if err := verifyNotDefaultRole(name); err != nil {
-		return err
-	}
+
 	// Verify storage constraints.
 	if err := ds.verifyRoleNameExists(ctx, name); err != nil {
 		return err
 	}
 
+	role, _, err := ds.roleStorage.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+	if err := verifyNotDefaultRole(role); err != nil {
```

Comment:
I think this check can be unified with `verifyRoleNameExists` - then we won't have to call `ds.roleStorage.Get` twice